src/views/PatronsPreRegistrationListContainer/hooks/useProcessPreRegisteredUser.js

```diff
@@ -0,0 +1,92 @@
+import { useMutation } from 'react-query';
+import { useHistory } from 'react-router-dom';
+import { FormattedMessage } from 'react-intl';
+
+import { useOkapiKy, useCallout } from '@folio/stripes/core';
+
+import { PATRON_PREREGISTRATIONS_API, PERMISSIONS_API } from '../../../constants';
+import { navigateToUserView } from '../../../utils/utils';
+
+const getFullName = (user) => {
+  let firstName = user?.generalInfo?.firstName ?? '';
+  const lastName = user?.generalInfo?.lastName ?? '';
+  const middleName = user?.generalInfo?.middleName ?? '';
+  const preferredFirstName = user?.generalInfo?.preferredFirstName ?? '';
+
+  if (preferredFirstName && firstName) {
+    firstName = `${preferredFirstName} (${firstName})`;
+  }
+
+  return `${lastName}${firstName ? ', ' : ' '}${firstName} ${middleName}`;
+};
+
+// this hook is used to create a new user from pre-registration record
+const useProcessPreRegisteredUser = () => {
+  const callout = useCallout();
+  const history = useHistory();
+  const ky = useOkapiKy();
+
+  const mergeOrCreateUser = (user) => {
+    return ky.put(`${PATRON_PREREGISTRATIONS_API}/${user.id}/mergeOrCreateUser`)
+      .json()
+      .then(({ userId }) => {
+        const name = getFullName(user);
+        callout.sendCallout({
+          message: <FormattedMessage
+            id="ui-users.stagingUser.createUser"
+            values={{ name }}
+          />,
+        });
+        return userId;
+      })
+      .catch((error) => {
+        throw new Error('Failed to merge or create user: ' + error);
+      });
+  };
+
+  const createPermissionUser = (userId) => {
+    return ky.post(PERMISSIONS_API, { json: { userId, permissions: [] } })
+      .catch((error) => {
+        throw new Error('Failed to create permission user: ' + error);
+      });
+  };
+
+
+  const handleError = (error) => {
+    // eslint-disable-next-line no-console
+    console.error(error);
```

Comment:
any reason for console?

Reply:
Yes, it is needed as per our PO.

Reply:
PO says you need to put a response error in console? I don't believe
what is the purpose? do regular users look at dev tools?
---
what PO asks you
```
Error handling.

If an error occurs while creating the new user record, the system should display an error toast message:
"Something went wrong. Please try again later."
```
I don't see anything here about logging in console

Reply:
Ah, then the console is merely following orders – straight from the PO’s mouth to its logs!

I am not here to bring in any belief. if you have any concerns, please talk to our POs Irina Poklets and Amelia Sutton.

Reply:
I don't want to talk to your POs, I review your PR not to be redirected to someone else :) in tickets there are no any requirements to add any logs to console - if they ask to do it, ask them to describe it in the ticket - but it will produce more questions why POs request something to be logged in console in additional to user friendly error handling